lib/plugins/menu/custom-events.js

```diff
@@ -0,0 +1,21 @@
+export const dispatchSelectCustomEvent = (select) => {
+  select.dispatchEvent(
+    new CustomEvent("visualEditorSelectChange", {
+      bubbles: true,
+      detail: {
+        selectText: select.options[select.selectedIndex].text,
+      },
+    }),
+  );
+};
+
+export const dispatchButtonCustomEvent = (button) => {
+  button.dispatchEvent(
+    new CustomEvent("visualEditorButtonClick", {
+      bubbles: true,
+      detail: {
+        buttonText: button.title,
```

Comment:
Further to the above about what we should expose to event consumers, maybe we should be including a few more things such as the ID. Although, when do we start thinking actually, it would have made more sense just to let the change/click events bubble up and catch the raw events from Whitehall... 😬